torch/csrc/generic/Tensor.cpp

```diff
@@ -816,6 +816,8 @@ PyTypeObject THPTensorStatelessType = {
 void THPTensor_(initCopyMethods)()
 {
   auto& h = THTensor_(copy_functions);
+  // copy from same type
+  THPInsertCopyFunction(h, &THTensor_(copy));
```

Comment:
`THTensor_(copy)` is the optimized one; THTensor_(copyXXX) is the slow "different types" version that can't do `memcpy`